corehq/blobs/management/commands/run_blob_migration.py

```diff
@@ -77,6 +77,15 @@ def add_argument(*args, **kw):
                 "parameter. Example value: 20180109-20190109"
             ),
         )
+        add_argument(
+            '--process_day_by_day',
+            action='store_true',
+            default=False,
+            help=(
+                "Run migration for each day in the given date-range separately "
+                "to allow cancelling and resuming on any day. Only applicable with date-range option"
```

Comment:
Curious why this was necessary? This may be less efficient than running it on the whole date range, and could result in some records not being migrated if date calculations are not done carefully.

The migration is already designed to be resumable, so it can be stopped and restarted repeatedly and it should pick up where it left of when restarted. If you're not seeing this behavior then there may be a bug (and we should track it down and fix it). I think this new option should be removed.

Reply:
I haven't tested ability to resume, but we need to be able to stop a bigger date range process and be able to split it into two separate processes when needed, I think that's not possible right now, is it?

In terms of some records not showing up, looks like `date_range` is inclusive, so `process_day_by_day` process for each day as a separate date-range, that should include all data, right?

In terms of efficiency, this provides ability to scale to multiple processes on fly which is useful on ICDS. Below is the migration process I am trying to follow.

There are around 32 CPUs that we could use, so, I am gonna split the 2018-2019 into ~32 date ranges and run them with `--process_day_by_day`. Since the data is not equally distributed on all months, some processes would finish quicker than other. Then, we can split other processes into two.  Happy to change the process, if there is a better way. (I was thinking to put all dates into a SQL table and then have processes process one date after other from the table which is more automatic, but requires more effort to get right) This was easiest.

Reply:
> we need to be able to stop a bigger date range process and be able to split it into two separate processes when needed, I think that's not possible right now, is it?

Interesting. You're correct, there is no support for that mode of operation right now.

> Since the data is not equally distributed on all months, some processes would finish quicker than other. Then, we can split other processes into two.

I'm concerned that the "split other processes into two" part is going to be tricky. Have you worked out a process for that part yet?